modules/consul-cluster/variables.tf

```diff
@@ -35,194 +40,223 @@ variable "user_data" {
+  type        = bool
   default     = false
 }
 
 variable "root_volume_type" {
   description = "The type of volume. Must be one of: standard, gp2, or io1."
+  type        = string
   default     = "standard"
 }
 
 variable "root_volume_size" {
   description = "The size, in GB, of the root EBS volume."
+  type        = number
   default     = 50
 }
 
 variable "root_volume_delete_on_termination" {
   description = "Whether the volume should be destroyed on instance termination."
+  type        = bool
   default     = true
 }
 
 variable "wait_for_capacity_timeout" {
   description = "A maximum duration that Terraform should wait for ASG instances to be healthy before timing out. Setting this to '0' causes Terraform to skip all Capacity Waiting behavior."
+  type        = string
   default     = "10m"
 }
 
 variable "service_linked_role_arn" {
   description = "The ARN of the service-linked role that the ASG will use to call other AWS services"
-  default     = ""
+  type        = string
+  default     = null
 }
 
 variable "health_check_type" {
   description = "Controls how health checking is done. Must be one of EC2 or ELB."
+  type        = string
   default     = "EC2"
 }
 
 variable "health_check_grace_period" {
   description = "Time, in seconds, after instance comes into service before checking health."
+  type        = number
   default     = 300
 }
 
 variable "instance_profile_path" {
   description = "Path in which to create the IAM instance profile."
+  type        = string
   default     = "/"
 }
 
 variable "server_rpc_port" {
   description = "The port used by servers to handle incoming requests from other agents."
+  type        = number
   default     = 8300
 }
 
 variable "cli_rpc_port" {
   description = "The port used by all agents to handle RPC from the CLI."
+  type        = number
   default     = 8400
 }
 
 variable "serf_lan_port" {
   description = "The port used to handle gossip in the LAN. Required by all agents."
+  type        = number
   default     = 8301
 }
 
 variable "serf_wan_port" {
   description = "The port used by servers to gossip over the WAN to other servers."
+  type        = number
   default     = 8302
 }
 
 variable "http_api_port" {
   description = "The port used by clients to talk to the HTTP API"
+  type        = number
   default     = 8500
 }
 
 variable "dns_port" {
   description = "The port used to resolve DNS queries."
+  type        = number
   default     = 8600
 }
 
 variable "ssh_port" {
   description = "The port used for SSH connections"
+  type        = number
   default     = 22
 }
 
 variable "tags" {
   description = "List of extra tag blocks added to the autoscaling group configuration. Each element in the list is a map containing keys 'key', 'value', and 'propagate_at_launch' mapped to the respective values."
-  type        = "list"
-  default     = []
+  type        = map(string)
+  default     = {}
 }
```

Comment:
See my comment above. I had assumed this was a change to allow the usage of:

```
tags = {
  key_for_tag1 = value_for_tag1
  key_for_tag2 = value_for_tag2
}
```
but it doesn't look like that is the case?

If this is what you intended, then the examples need to be updated + the docs for the variable. It would be good to document this change in the release notes as well.